test/components/preprocessors/test_markdown_header_splitter.py

```diff
@@ -0,0 +1,324 @@
+    with pytest.raises(ValueError, match="only works with text documents"):
+        splitter.run(documents=docs)
+
+
+def test_empty_document_list():
+    """Test handling of an empty document list."""
+    splitter = MarkdownHeaderSplitter()
+    result = splitter.run(documents=[])
+    assert result["documents"] == []
+
+
+def test_invalid_secondary_split_at_init():
+    """Test that an invalid secondary split type raises an error at initialization time."""
+    with pytest.raises(ValueError, match="split_by must be one of"):
+        MarkdownHeaderSplitter(secondary_split="invalid_split_type")
+
+
+def test_invalid_split_parameters_at_init():
+    """Test invalid split parameter validation at initialization time."""
+    # Test split_length validation
+    with pytest.raises(ValueError, match="split_length must be greater than 0"):
+        MarkdownHeaderSplitter(secondary_split="word", split_length=0)
+
+    # Test split_overlap validation
+    with pytest.raises(ValueError, match="split_overlap must be greater than or equal to 0"):
+        MarkdownHeaderSplitter(secondary_split="word", split_overlap=-1)
+
+
+def test_empty_content_handling():
+    """Test handling of documents with empty content."""
+    splitter_skip = MarkdownHeaderSplitter()  # skip empty documents by default
+    docs = [Document(content="")]
+    result = splitter_skip.run(documents=docs)
+    assert len(result["documents"]) == 0
+
+    splitter_no_skip = MarkdownHeaderSplitter(skip_empty_documents=False)
+    docs = [Document(content="")]
+    result = splitter_no_skip.run(documents=docs)
+    assert len(result["documents"]) == 1
+
+
+def test_split_id_sequentiality_primary_and_secondary(sample_text):
+    # Test primary splitting
+    splitter = MarkdownHeaderSplitter(keep_headers=False)
+    docs = [Document(content=sample_text)]
+    result = splitter.run(documents=docs)
+    split_docs = result["documents"]
+
+    # Test number of documents
+    assert len(split_docs) == 5
+
+    # Check that split_ids are sequential
+    split_ids = [doc.meta["split_id"] for doc in split_docs]
+    assert split_ids == list(range(len(split_ids)))
+
+    # Test secondary splitting
+    splitter = MarkdownHeaderSplitter(secondary_split="word", split_length=3)
+    docs = [Document(content=sample_text)]
+    result = splitter.run(documents=docs)
+    split_docs = result["documents"]
+
+    # Test number of documents
+    assert len(split_docs) == 12
+
+    split_ids = [doc.meta["split_id"] for doc in split_docs]
+    assert split_ids == list(range(len(split_ids)))
+
+    # Test with multiple input documents
+    docs = [Document(content=sample_text), Document(content="# Another Header\nSome more content here.")]
+    result = splitter.run(documents=docs)
+    split_docs = result["documents"]
+
+    # Test number of documents
+    assert len(split_docs) == 14
+
+    split_ids = [doc.meta["split_id"] for doc in split_docs]
+    assert split_ids == list(range(len(split_ids)))
```

Comment:
This should be updated once this comment https://github.com/deepset-ai/haystack/pull/9660/files#r2432221323 is addressed